BuildCraft for 1.16.5 Forge

CalenXwX opened the pull request:
WuHoo~
Now seems BC works fine in 1.16.5 Forge.

The code is similar to the PR of 1.18.2, with some changes in class names and others.

Expecting BC's rebirth :)

awa

CalenXwX committed:
1.16.5!

CalenXwX committed:
Correct reflection

CalenXwX committed:
Switched submodule links and updated mods.toml

CalenXwX committed:
Updated submodule

CalenXwX committed:
Updated expression module

CalenXwX committed:
Fixed TileHeatExchange

Fixed:
1. Lava was not able to be used as coolable fluid.
2. Fluid in [Start] section will not be stored to save data. (a bug from 1.12.2)

CalenXwX committed:
Updated submodule

Leclowndu93150 commented:
bro's doing great work

CalenXwX committed:
Fixed PipeStripes extending

CalenXwX committed:
Fixed: builder rendering & builder pipe rotation & required items of plugs

1. fixed rendering position of builder's path
2. plugs on pipes now can rotate with the rotation of builder block (a bug from 1.12.2)
3. required items of pipe blocks now contains plugs

CalenXwX committed:
Fix: builder pipe rotation

CalenXwX committed:
Fix: dependencies

Now the dependencies [vecmath, trove4j] can be loaded correctly... but the jar files will be copied to GAME_DIR/buildcraft_dependencies/
Seems 1.16.5 forge does not support jar in jar!

CalenXwX committed:
Fix: builder required material in TileEntity's storage

CalenXwX committed:
FIX: trove4j dependency

CalenXwX committed:
Fix: IVertexBuilder class cast

mistermodcreator commented:
True though.

Maybe someone could release this as a fork of buildcraft on CurseForge, so people can finally use it in their modpacks? Seems like this pull request isn't going to be merged anytime soon.

CalenXwX committed:
Updated gitsubmodule

CalenXwX committed:
Robotics!

CalenXwX committed:
Fixed: Ledger owner face skin

CalenXwX committed:
Update from latest commits of 1.12.2: Engine Pipe RF PlugTimer

CalenXwX committed:
Fixed registration

CalenXwX committed:
Update daizuli item pipe: selectable logic(strict direction / direction priority)

And some other simple updates

CalenXwX committed:
Add configs to enable/disable facades, fix NPE when facade item disabled and fix gui player head sprite loading

CalenXwX committed:
Fix #21 tags

CalenXwX committed:
Fix MJ-RF conversion & Make programming recipe input item modifiable

CalenXwX committed:
Fix Laser & Quarry rendering and Quarry debug message

CalenXwX committed:
Fix some bugs

Fixed NPE when BC modules using BCLibRegistries inisialize earlier than BCLib.
Fixed GuiArchitectTable closes when user type <E> in the edit box.
Fixed the edit box of GuiReplacer was not clickable. (but the edit box is still useless)
Fixed the edit box in GuiList was not editable.
Fixed Create mod's HosePulley could not enter and drain oil blocks.
Fixed "\u" in config jsons.
Pipe's map colours are not changed in 1.16.5 because seems forge does not support this.

CalenXwX committed:
Fix build
